RIOT-OS/RIOT
# doc: update to jquery 3.5.1

### Contribution description

This PR replaces our outdated jquery with current jquery 3.5.1 and jquery-migrate 3.3.0.

The reason for doing this is that github greets everyone on https://github.com/RIOT-OS/RIOT with a security vulnerability warning.

### Testing procedure

I have no clue about jquery, so please double check and test.
Take a look at the output, make sure everything works and looks as before.

Search doesn't work, but it is broken for me on master, too.



### Issues/PRs references
